### ISSUE TITLE
Feature request: capture invisible window

### ISSUE THREAD

**Issue by DjulLau** [nwjs/nw.js#6826]
Actually it's not possible to capture a window if it's not visible. See #4814. 
`capturePage()` would fail with :
```
Error in response to nw.currentWindowInternal.capturePageInternal: TypeError: Cannot read property 'replace' of undefined
Unchecked runtime.lastError while running nw.currentWindowInternal.capturePageInternal: Failed to capture tab: unknown error
```
The app we are working on uses `capturePage()` intensively to make preview thumbs of dynamic content while installing/updating. 
We don't want the end user to see this process (he's only provided a progress/info dialog). 
Here's how it works so far:
*  Make the main window `AlwaysOnTop`
*  Open a new capture window
*  Give it the exact same size/position as the main window (so it's not visible)
*  Load content and do capture using `capturePage()`
*  Close the capture window
*  Remove `AlwaysOnTop` on the main window

It does the job **but** this mean user can't do anything else while installing/updating which is particulary annoying when there is a lot of content (full process can take like 10 minutes).
So we would be very interested to be able to do this, in the background, without the capture window needing to be visible. 
If anyone can help I think we could put a bounty on this feature (don't know how) !

Regards,

**Comment by averzen** [comment]
Agree. I think capturePage is only useful with hidden windows as like some render process.
We use a hidden window to capture pages thumbnails without any unnecessary elements on top of it. And in last update it was broken, so it needs to be showed or focused.
